### PR TITLE
Azure: Added support to ephemeral OS Disk (Preview)

### PR DESCRIPTION
**Pull request by iamshital**
Now, you can deploy VMs with ephemeral OS disk using following xml tag :
```
<AdditionalHWConfig>
  <DiskType>Managed</DiskType>
  <OSDiskType>Ephemeral</OSDiskType>
</AdditionalHWConfig>
```
Note, You can not use Unmanaged disk + Ephemeral disk configuration.

Validations needed -
**Using default image (ARMImage)**
- [x] Single VM - Managed OS DIsk - No data disk
- [x] Single VM - Managed OS DIsk - With managed data disks
- [x] Single VM - Managed Ephemeral OS DIsk - No data disk
- [x] Single VM - Managed Ephemeral OS DIsk - With managed data disks

**Using custom VHD (Captured Image)**
- [x] Single VM - Managed OS DIsk - No data disk
- [x] Single VM - Managed OS DIsk - With managed data disks
- [x] Single VM - Managed Ephemeral OS DIsk - No data disk
- [x] Single VM - Managed Ephemeral OS DIsk - With managed data disks

### CODE REVIEW

File: Libraries/Azure.psm1 [hunk 1]
```diff
@@ -1980,13 +2010,24 @@ foreach ( $newVM in $RGXMLData.VirtualMachine)
...
                             Add-Content -Value "$($indents[6])^name^: ^$vmName-OSDisk^," -Path $jsonFile
-                            Add-Content -Value "$($indents[6])^createOption^: ^FromImage^," -Path $jsonFile
                             Add-Content -Value "$($indents[6])^managedDisk^: " -Path $jsonFile
                             Add-Content -Value "$($indents[6]){" -Path $jsonFile
                                 Add-Content -Value "$($indents[7])^storageAccountType^: ^$StorageAccountType^" -Path $jsonFile
-                            Add-Content -Value "$($indents[6])}" -Path $jsonFile
-                            LogMsg "Added managed OS disk : $vmName-OSDisk"
-
+                            Add-Content -Value "$($indents[6])}," -Path $jsonFile
+                        if ($UseEphemeralOSDisk)
```
Comment (iamshital) [inline Libraries/Azure.psm1]: The indents here are adjusted to simplify creation of JSON file. It's not a bug.

**Comment by iamshital** [comment]
Not to merge until further notification.

**Comment by iamshital** [comment]
> The change looks OK, please make sure it's well tested. @iamshital

Yes @lizzha I've added necessary validations, in the PR description. 
If you have any, you can add them there.

**Comment by juhlee-microsoft** [comment]
@iamshital @lizzha It looks ok to me, but @iamshital  is doing testing for all cases. I suggest to merge after the all valid results.

**Comment by juhlee-microsoft** [comment]
@iamshital It's been 9 days passed after my last comment. if you do not work on this any more, let me know. I will close it on Monday.

**Comment by iamshital** [comment]
@juhlee-microsoft , I'll have it validated for all the scenarios by tomorrow.

**Comment by iamshital** [comment]
@juhlee-microsoft All checks are done.

**Comment by iamshital** [comment]
@juhlee-microsoft , @lizzha At this point, I think the json generator code is mature enough to re-foctor according to coding style. 
Hence, I've opened another PR to fix all the formatting issues in Azure.psm1 file in PR #336

**Pull request merged**